code/modules/ftl/machinery/ftl_shieldgen.dm

```diff
@@ -93,6 +125,14 @@
 	update_icon()
 	update_active()
 
+/obj/machinery/ftl_shieldgen/attackby(obj/item/weapon/W, mob/user, params)
```

Comment:
Im not really sure how this part works, can you elaborate? besides that. this PR should be fine.

Reply:
If you mean the attackby(...), then I added it, so that the parts could be exchanged, and so that the shieldgen could be deconstructed. And now I just noticed the deconstruction screwdriver thingy with ore_redemption, huh.